source/exe/main.cc

```diff
@@ -19,24 +22,21 @@
  *
  * This should be used to do setup tasks specific to a particular site's
  * deployment such as initializing signal handling.  It calls main_common
- * after setting up command line options and the hot restarter.
+ * after setting up command line options.
  */
 int main(int argc, char** argv) {
 #ifdef ENVOY_HANDLE_SIGNALS
   // Enabled by default. Control with "bazel --define=signal_trace=disabled"
   Envoy::SignalAction handle_sigs;
 #endif
 
-  Envoy::OptionsImpl options(argc, argv, Envoy::Server::SharedMemory::version(),
-                             spdlog::level::warn);
+#ifdef ENVOY_HOT_RESTART
+  const std::string& shared_mem_version = Envoy::Server::SharedMemory::version();
+#else
+  const std::string& shared_mem_version = "disabled";
```

Comment:
technically you are taking a reference to a temporary here. I would probably just define on the previous line and then take a reference to it.